internal/graph/cached_resolver_test.go

```diff
@@ -374,17 +374,20 @@ func TestResolveCheckFromCache(t *testing.T) {
 			test.setInitialResult(mockResolver, initialReq)
 
 			// expect first call to result in actual resolve call
-			dut := NewCachedCheckResolver(mockResolver,
-				WithMaxCacheSize(10))
+			dut := NewCachedCheckResolver(WithMaxCacheSize(10))
 			defer dut.Close()
 
+			dut.SetDelegate(mockResolver)
+
 			_, _ = dut.ResolveCheck(ctx, initialReq)
 
 			test.setTestExpectations(mockResolver, test.subsequentReq)
 
-			dut2 := NewCachedCheckResolver(dut, WithExistingCache(dut.cache))
+			dut2 := NewCachedCheckResolver(WithExistingCache(dut.cache))
 			defer dut2.Close()
 
+			dut2.SetDelegate(dut)
+
 			actualResult, err := dut2.ResolveCheck(ctx, test.subsequentReq)
 			require.Equal(t, result.Allowed, actualResult.Allowed)
 			require.NoError(t, err)
```

Comment:
I suppose this test ensures that if you pass in `WithExistingCache` it actually uses it instead of constructing an internal one and using it instead AND that we return the previously computed cache result. It's possible for a bug to creep into the CachedCheckResolver where the `dut2` could return `allowed: false` even though the initial request returned `allowed: true`. It's unlikely, but I guess that's what we're protected against.

The `test.setTestExpectations(mockResolver, test.subsequentReq)` ensures we don't end up calling `ResolveCheck` a subsequent time, but it doesn't ensure that we get the prior computed result back. So having this covers our bases a bit better.